CoreUtilities/src/au/gov/asd/tac/constellation/utilities/gui/JMultiChoiceComboBoxMenu.java

```diff
@@ -238,6 +234,22 @@ public final String getToolTipText() {
         return button.getToolTipText();
     }
 
+    protected Set<ListSelectionListener> getListeners() {
+        return listeners;
+    }
+
+    protected JButton getButton() {
+        return button;
+    }
+
+    protected void setButton(JButton button) {
+        this.button = button;
+    }
```

Comment:
is there a way your unit tests can be written such that this setter isn't needed? I tend to be ok with getters being added for unit testing but a little more reluctant to add setters if unit testing is their only use. Especially seeing as in this instance you also had to make `button` non-final in the process

Reply:
I couldn't find a way around this. The issue is because this code is in the constructor and the mock on the `JButton` doesn't apply unless you set it again. Do you have other ideas?

Reply:
given the button is created within the constructor, could you just use the getter to get the button and then call something like: `button.getActionListeners()[0].actionPerformed(null);`? (the parameter for actionPerformed isn't used in this case so you can pass null ok)

Reply:
The problem is `button.getLocationOnScreen()` which  relies on it being shown on the screen unless we mock. This is the error:
![image](https://user-images.githubusercontent.com/64826928/142572318-ab710c6d-a4fe-490c-a003-96c67a75fed0.png)